Invalid input error: Invalid resource references for experiment. Expect one namespace type with owner relationship

I deployed kubeflow from using [this kfdef](https://gist.github.com/reevejd/7116780665e871f043005df6781f6ab7) (uses manifests from [v1.1-branch](https://github.com/kubeflow/kubeflow/tree/v1.1-branch))

When trying to run a pipeline on an external, multi-user auth cluster I’m getting a modal pop-up with this error:
<img width="487" alt="Screen Shot 2020-11-09 at 5 20 14 PM" src="https://user-images.githubusercontent.com/12180828/98608986-5af1f580-22ba-11eb-9cea-8365eeda7024.png">

The relevant parts of the JupyterLab logs seem to be:
```
Validate experiment request failed.: Invalid input error: Invalid resource references for experiment. Expect one namespace type with owner relationship. Got:[]","message":"Validate experiment request failed.: Invalid input error: Invalid resource references for experiment.
```
[The full trace is in this gist](https://gist.github.com/reevejd/01cdacb743215969c279908fd68723c0).

Elyra successfully creates the pipeline, and if I then log in to the kubeflow GUI and manually create an experiment from the pipeline Elyra created, it works as expected.

I notice the same error message in a [kubeflow/pipelines issue here](https://github.com/kubeflow/pipelines/issues/4440) but it seems to only apply only to applications using the in-cluster endpoint, which isn’t the case for me.

Hi @akchinSTC, thanks for the update. Unfortunately haven't been able to build from your branch to test it out. Running `make clean install` gives me a `not a valid npm package` error because of an underlying typescript error:
```
src/CodeSnippetWidget.tsx(445,11): error TS2322: Type '{ children: Element; displayName: string; tooltip: any; actionButtons: IMetadataActionButton[]; onExpand: () => void; onMouseDown: (event: any) => void; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<ExpandableComponent> & Readonly<IExpandableComponentProps> & Readonly<...>'.
```
running `make docker-image` also errors out, but with a different error message:
```
#9 8.995 ERROR: Cannot uninstall 'terminado'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
#9 ERROR: executor failed running [/bin/sh -c pip install --upgrade pip jupyterlab notebook]: runc did not terminate sucessfully
------
 > [4/9] RUN pip install --upgrade pip jupyterlab notebook:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = failed to build LLB: executor failed running [/bin/sh -c pip install --upgrade pip jupyterlab notebook]: runc did not terminate sucessfully
make: *** [docker-image] Error 1
```
If there's an easy fix for the above errors, great, I can do that. Otherwise if you publish a dev docker image I can check if your fix works for me.

--- a/elyra/metadata/schemas/kfp.json
+++ b/elyra/metadata/schemas/kfp.json
@@ -57,6 +57,16 @@
             "category": "Kubeflow Pipelines"
           }
         },
+        "user_namespace": {
+          "title": "Kubeflow Pipelines User Namespace",
+          "description": "The Kubeflow Pipelines user namespace used to create experiments",
+          "type": "string",
+          "pattern": "^[a-z0-9][-a-z0-9]*[a-z0-9]$",
+          "maxLength": 63,
+          "uihints": {
+            "category": "Kubeflow Pipelines"
+          }
+        },
         "cos_endpoint": {
           "title": "Cloud Object Storage Endpoint",
           "description": "The Cloud Object Storage endpoint",

--- a/elyra/pipeline/processor_kfp.py
+++ b/elyra/pipeline/processor_kfp.py
@@ -53,6 +53,7 @@ class KfpPipelineProcessor(PipelineProcessor):
         # TODO: try to encapsulate the info below
         api_username = runtime_configuration.metadata.get('api_username')
         api_password = runtime_configuration.metadata.get('api_password')
+        user_namespace = runtime_configuration.metadata['user_namespace']
 
         self.log_pipeline_info(pipeline_name, "submitting pipeline")
         with tempfile.TemporaryDirectory() as temp_dir:
@@ -99,7 +100,10 @@ class KfpPipelineProcessor(PipelineProcessor):
                 else:
                     raise lve
 
-            run = client.run_pipeline(experiment_id=client.create_experiment(pipeline_name).id,
+            experiment = client.create_experiment(name=pipeline_name,
+                                                  namespace=user_namespace)
+
+            run = client.run_pipeline(experiment_id=experiment.id,
                                       job_name=timestamp,
                                       pipeline_id=kfp_pipeline.id)
 

--- a/elyra/pipeline/processor_kfp.py
+++ b/elyra/pipeline/processor_kfp.py
@@ -51,7 +51,7 @@ class KfpPipelineProcessor(PipelineProcessor):
         cos_bucket = runtime_configuration.metadata['cos_bucket']
 
         user_namespace = None
-        if runtime_configuration.metadata['user_namespace']:
+        if runtime_configuration.metadata.get('user_namespace'):
             user_namespace = runtime_configuration.metadata['user_namespace']
 
         # TODO: try to encapsulate the info below

--- a/elyra/pipeline/processor_kfp.py
+++ b/elyra/pipeline/processor_kfp.py
@@ -50,9 +50,7 @@ class KfpPipelineProcessor(PipelineProcessor):
         cos_endpoint = runtime_configuration.metadata['cos_endpoint']
         cos_bucket = runtime_configuration.metadata['cos_bucket']
 
-        user_namespace = None
-        if runtime_configuration.metadata.get('user_namespace'):
-            user_namespace = runtime_configuration.metadata['user_namespace']
+        user_namespace = runtime_configuration.metadata.get('user_namespace')
 
         # TODO: try to encapsulate the info below
         api_username = runtime_configuration.metadata.get('api_username')

--- a/elyra/pipeline/processor_kfp.py
+++ b/elyra/pipeline/processor_kfp.py
@@ -50,10 +50,13 @@ class KfpPipelineProcessor(PipelineProcessor):
         cos_endpoint = runtime_configuration.metadata['cos_endpoint']
         cos_bucket = runtime_configuration.metadata['cos_bucket']
 
+        user_namespace = None
+        if runtime_configuration.metadata['user_namespace']:
+            user_namespace = runtime_configuration.metadata['user_namespace']
+
         # TODO: try to encapsulate the info below
         api_username = runtime_configuration.metadata.get('api_username')
         api_password = runtime_configuration.metadata.get('api_password')
-        user_namespace = runtime_configuration.metadata['user_namespace']
 
         self.log_pipeline_info(pipeline_name, "submitting pipeline")
         with tempfile.TemporaryDirectory() as temp_dir:

